homeassistant/components/dyson/fan.py

```diff
@@ -198,24 +274,15 @@ def __init__(self, device):
     def set_speed(self, speed: str) -> None:
         """Set the speed of the fan. Never called ??."""
         _LOGGER.debug("Set fan speed to: %s", speed)
-
-        if speed == FanSpeed.FAN_SPEED_AUTO.value:
-            self._device.set_configuration(fan_mode=FanMode.AUTO)
-        else:
-            fan_speed = FanSpeed(f"{int(speed):04d}")
-            self._device.set_configuration(fan_mode=FanMode.FAN, fan_speed=fan_speed)
+        self._device.set_configuration(
```

Comment:
Check first that the speed is in the allowed speed values and raise a ValueError if it's not.